### PR TITLE
work on BSM operator and inversions

### PR DESCRIPTION
**Pull request by kostrzewa**
* Add support for multiple external fields per gauge (in general, scalar fields for the BSM operator)
* in op_invert, have the BSM operator pass the number of iterations and the final residual squarenorm to the operator struct

### CODE REVIEW

File: invert.c [hunk 1]
```diff
@@ -323,32 +323,33 @@ int main(int argc, char *argv[])
...
-	  exit(-2);
-	}
-	
-	if (g_cart_id == 0) {
-	  printf("# Finished reading scalar field.\n");
-	  fflush(stdout);
-	}
-      }
-    }
+    // the scalar field is generated / read below
```
Comment (kostrzewa) [inline invert.c]: oops, sorry, I'll remove this mess!

**Comment by kostrzewa** [comment]
In addition to the commented out section, the logic is not correct... right now the code will generate new volume sources for each external field, which is obviously wrong...

**Comment by kostrzewa** [comment]
Ugh, just realized that the scalar field I/O is not MPI aware...

**Comment by urbach** [comment]
in principle there speaks nothing against a new random source per scalar field, does it?

**Comment by kostrzewa** [comment]
Well, if we want to see the error scaling in three dimensions, then the source should be the same, no?

**Comment by kostrzewa** [comment]
Also, the location of the scalar reading / generating should now be correct also for using multiple scalar fields with point sources.

**Pull request merged**